fiftyone/core/collections.py

```diff
@@ -916,26 +916,45 @@ def _get_per_sample_frames_bytes(self):
         results = self._aggregate(pipeline=pipeline, attach_frames=True)
         return {str(r["_id"]): r["size_bytes"] for r in results}
 
+    def _get_first(self, limit=1, field="_id", reverse=False):
+        direction = -1 if reverse else 1
+        pipeline = [{"$sort": {field: direction}}, {"$limit": limit}]
```

Comment:
It should be noted that the implementation of `view.last()` is now significantly different than it used to be.

Previously it would run two pipelines: `view_pipeline + [{"$count": n}]` to get the count followed by `view_pipeline + [{"$skip": n - 1}]` to get the last sample.

Now it runs a single pipeline: `view_pipeline + [{"$sort": {"_id": -1}}, {"$limit": 1}]` that directly grabs the last sample.

`$count` and `$skip` require collection scans, which was definitely not ideal and is the key thing that this PR is addressing in the first place. However, at least they probably(?) don't require too much DB memory.

I'm uncertain how performant `[{$sort: _id}, {$limit: 1}]` is when appended to a potentially complex pipeline. I worry that the sorting might require more memory?